In [1]:
import pandas as pd
import numpy as np

### Cleaning grades_info Dataset

In [2]:
#Reading in the data for grades
grade_info = pd.read_csv("../raw_data/grade_info.csv")

In [3]:
grade_info.head(20)

grade sub_grade
0      B        B2
1      C        C4
2      C        C5
3      C        C1
4      B        B5
5      A        A4
6      E        E1
7      F        F2
8      C        C3
9      B        B1
10     D        D1
11     A        A1
12     B        B3
13     B        B4
14     C        C2
15     D        D2
16     A        A3
17     A        A5
18     D        D5
19     A        A2

In [4]:
#Getting dimensions
grade_info.shape

(36, 2)

In [5]:
#checking for missing values
grade_info.isna().sum()

grade        1
sub_grade    1
dtype: int64

In [6]:
grade_info.isna()

grade  sub_grade
0   False      False
1   False      False
2   False      False
3   False      False
4   False      False
5   False      False
6   False      False
7   False      False
8   False      False
9   False      False
10  False      False
11  False      False
12  False      False
13  False      False
14  False      False
15  False      False
16  False      False
17  False      False
18  False      False
19  False      False
20  False      False
21  False      False
22  False      False
23  False      False
24  False      False
25  False      False
26  False      False
27  False      False
28  False      False
29  False      False
30  False      False
31  False      False
32  False      False
33  False      False
34  False      False
35   True       True

In [7]:
grade_info.tail()

grade sub_grade
31     G        G2
32     G        G1
33     F        F5
34     G        G5
35   NaN       NaN

In [8]:
grade_info.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 36 entries, 0 to 35
Data columns (total 2 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   grade      35 non-null     object
 1   sub_grade  35 non-null     object
dtypes: object(2)
memory usage: 704.0+ bytes


The last row has NaN values.

In [9]:
#Removing the last row with missing values and saving as a new dataframe "grades_clean"
grades_clean = grade_info.drop([0, 35]).copy()
grades_clean

grade sub_grade
1      C        C4
2      C        C5
3      C        C1
4      B        B5
5      A        A4
6      E        E1
7      F        F2
8      C        C3
9      B        B1
10     D        D1
11     A        A1
12     B        B3
13     B        B4
14     C        C2
15     D        D2
16     A        A3
17     A        A5
18     D        D5
19     A        A2
20     E        E4
21     D        D3
22     D        D4
23     F        F3
24     E        E3
25     F        F4
26     F        F1
27     E        E5
28     G        G4
29     E        E2
30     G        G3
31     G        G2
32     G        G1
33     F        F5
34     G        G5

In [10]:
#Writing grades_clean to csv and saving in clean_data files.
grades_clean.to_csv("../clean_data/grades_clean.csv")

### Reading in data dictionary.

In [11]:
#Reading in data dictionary
data_dictionary = pd.read_csv("../raw_data/LCDataDictionary.csv")

#Ensuring description column can be read in full.
pd.set_option("display.max_colwidth", None)
pd.set_option('display.max_rows', 120)

data_dictionary

LoanStatNew  \
0                    acc_now_delinq   
1              acc_open_past_24mths   
2                        addr_state   
3                          all_util   
4                        annual_inc   
5                  annual_inc_joint   
6                  application_type   
7                       avg_cur_bal   
8                    bc_open_to_buy   
9                           bc_util   
10         chargeoff_within_12_mths   
11          collection_recovery_fee   
12       collections_12_mths_ex_med   
13                      delinq_2yrs   
14                      delinq_amnt   
15                             desc   
16                              dti   
17                        dti_joint   
18                 earliest_cr_line   
19                       emp_length   
20                        emp_title   
21                  fico_range_high   
22                   fico_range_low   
23                      funded_amnt   
24                  funded_amnt_inv   
25                            grade   
26                   home_ownership   
27                               id   
28                          il_util   
29              initial_list_status   
30                           inq_fi   
31                     inq_last_12m   
32                   inq_last_6mths   
33                      installment   
34                         int_rate   
35                          issue_d   
36               last_credit_pull_d   
37             last_fico_range_high   
38              last_fico_range_low   
39                  last_pymnt_amnt   
40                     last_pymnt_d   
41                        loan_amnt   
42                      loan_status   
43                       max_bal_bc   
44                        member_id   
45               mo_sin_old_il_acct   
46             mo_sin_old_rev_tl_op   
47            mo_sin_rcnt_rev_tl_op   
48                   mo_sin_rcnt_tl   
49                         mort_acc   
50           mths_since_last_delinq   
51      mths_since_last_major_derog   
52           mths_since_last_record   
53               mths_since_rcnt_il   
54             mths_since_recent_bc   
55         mths_since_recent_bc_dlq   
56            mths_since_recent_inq   
57   mths_since_recent_revol_delinq   
58                     next_pymnt_d   
59            num_accts_ever_120_pd   
60                   num_actv_bc_tl   
61                  num_actv_rev_tl   
62                      num_bc_sats   
63                        num_bc_tl   
64                        num_il_tl   
65                    num_op_rev_tl   
66                    num_rev_accts   
67              num_rev_tl_bal_gt_0   
68                         num_sats   
69                 num_tl_120dpd_2m   
70                     num_tl_30dpd   
71               num_tl_90g_dpd_24m   
72               num_tl_op_past_12m   
73                         open_acc   
74                      open_acc_6m   
75                      open_il_12m   
76                      open_il_24m   
77                       open_il_6m   
78                      open_rv_12m   
79                      open_rv_24m   
80                        out_prncp   
81                    out_prncp_inv   
82                   pct_tl_nvr_dlq   
83                 percent_bc_gt_75   
84                      policy_code   
85                          pub_rec   
86             pub_rec_bankruptcies   
87                          purpose   
88                       pymnt_plan   
89                       recoveries   
90                        revol_bal   
91                       revol_util   
92                        sub_grade   
93                        tax_liens   
94                             term   
95                            title   
96                     tot_coll_amt   
97                      tot_cur_bal   
98                  tot_hi_cred_lim   
99                        total_acc   
100               total_bal_ex_mort   
101                    total_bal_il   
102    

### Cleaning lending_club_loans dataset

In [12]:
#Reading in the data for lending_club_loans
lending_club_loans = pd.read_csv("../raw_data/lending_club_loans.csv", low_memory = False)

In [13]:
lending_club_loans

id  member_id  loan_amnt  funded_amnt  funded_amnt_inv  \
0      1077501.0  1296599.0     5000.0       5000.0           4975.0   
1      1077430.0  1314167.0     2500.0       2500.0           2500.0   
2      1077175.0  1313524.0     2400.0       2400.0           2400.0   
3      1076863.0  1277178.0    10000.0      10000.0          10000.0   
4      1075358.0  1311748.0     3000.0       3000.0           3000.0   
...          ...        ...        ...          ...              ...   
42533    72176.0    70868.0     2525.0       2525.0            225.0   
42534    71623.0    70735.0     6500.0       6500.0              0.0   
42535    70686.0    70681.0     5000.0       5000.0              0.0   
42536        NaN        NaN        NaN          NaN              NaN   
42537        NaN        NaN        NaN          NaN              NaN   

            term int_rate  installment sub_grade                 emp_title  \
0      36 months   10.65%       162.87        B2                       NaN   
1      60 months   15.27%        59.83        C4                     Ryder   
2      36 months   15.96%        84.33        C5                       NaN   
3      36 months   13.49%       339.31        C1       AIR RESOURCES BOARD   
4      60 months   12.69%        67.79        B5  University Medical Group   
...          ...      ...          ...       ...                       ...   
42533  36 months    9.33%        80.69        B3                       NaN   
42534  36 months    8.38%       204.84        A5                       NaN   
42535  36 months    7.75%       156.11        A3                 Homemaker   
42536        NaN      NaN          NaN       NaN                       NaN   
42537        NaN      NaN          NaN       NaN                       NaN   

       ... num_tl_90g_dpd_24m num_tl_op_past_12m  pct_tl_nvr_dlq  \
0      ...                NaN                NaN             NaN   
1      ...                NaN                NaN             NaN   
2      ...                NaN                NaN             NaN   
3      ...                NaN                NaN             NaN   
4      ...                NaN                NaN             NaN   
...    ...                ...                ...             ...   
42533  ...                NaN                NaN             NaN   
42534  ...                NaN                NaN             NaN   
42535  ...                NaN                NaN             NaN   
42536  ...                NaN                NaN             NaN   
42537  ...                NaN                NaN             NaN   

      percent_bc_gt_75 pub_rec_bankruptcies tax_liens tot_hi_cred_lim  \
0                  NaN                  0.0       0.0             NaN   
1                  NaN                  0.0       0.0             NaN   
2                  NaN                  0.0       0.0             NaN   
3                  NaN                  0.0       0.0             NaN   
4                  NaN                  0.0       0.0             NaN   
...                ...                  ...       ...             ...   
42533              NaN                  NaN       NaN             NaN   
42534              NaN                  NaN       NaN             NaN   
42535              NaN                  NaN       NaN             NaN   
42536              NaN                  NaN       NaN             NaN   
42537              NaN                  NaN       NaN             NaN   

      total_bal_ex_mort total_bc_limit total_il_high_credit_limit  
0                   NaN            NaN                        NaN  
1                   NaN            NaN                        NaN  
2                   NaN            NaN                        NaN  
3                   NaN            NaN                        NaN  
4                   NaN            NaN                        NaN  
...                 ...            ...                        ...  
42533               NaN            NaN    

In [14]:
lending_club_loans.shape

(42538, 114)

42538 rows and 114 columns. There are quite a few columns which will be irrelevant in predicting loan defaults so these will be removed.  

In [15]:
#Taking a look at all variable names
lending_club_loans.iloc[0]

id                                                                                                  1.0775e+06
member_id                                                                                           1.2966e+06
loan_amnt                                                                                                 5000
funded_amnt                                                                                               5000
funded_amnt_inv                                                                                           4975
term                                                                                                 36 months
int_rate                                                                                                10.65%
installment                                                                                             162.87
sub_grade                                                                                                   B2
e

To begin with, I will remove the following columns:- zip_code, id, member_id, url, emp_title, issue_d & description.

In [16]:
loans_clean = lending_club_loans.drop(["id", "member_id", "zip_code", "url", 
                                       "emp_title", "issue_d", "desc"], axis = 1).copy()
loans_clean

loan_amnt  funded_amnt  funded_amnt_inv       term int_rate  \
0         5000.0       5000.0           4975.0  36 months   10.65%   
1         2500.0       2500.0           2500.0  60 months   15.27%   
2         2400.0       2400.0           2400.0  36 months   15.96%   
3        10000.0      10000.0          10000.0  36 months   13.49%   
4         3000.0       3000.0           3000.0  60 months   12.69%   
...          ...          ...              ...        ...      ...   
42533     2525.0       2525.0            225.0  36 months    9.33%   
42534     6500.0       6500.0              0.0  36 months    8.38%   
42535     5000.0       5000.0              0.0  36 months    7.75%   
42536        NaN          NaN              NaN        NaN      NaN   
42537        NaN          NaN              NaN        NaN      NaN   

       installment sub_grade emp_length home_ownership  annual_inc  ...  \
0           162.87        B2  10+ years           RENT     24000.0  ...   
1            59.83        C4   < 1 year           RENT     30000.0  ...   
2            84.33        C5  10+ years           RENT     12252.0  ...   
3           339.31        C1  10+ years           RENT     49200.0  ...   
4            67.79        B5     1 year           RENT     80000.0  ...   
...            ...       ...        ...            ...         ...  ...   
42533        80.69        B3   < 1 year           RENT    110000.0  ...   
42534       204.84        A5   < 1 year           NONE         NaN  ...   
42535       156.11        A3  10+ years       MORTGAGE     70000.0  ...   
42536          NaN       NaN        NaN            NaN         NaN  ...   
42537          NaN       NaN        NaN            NaN         NaN  ...   

      num_tl_90g_dpd_24m num_tl_op_past_12m pct_tl_nvr_dlq percent_bc_gt_75  \
0                    NaN                NaN            NaN              NaN   
1                    NaN                NaN            NaN              NaN   
2                    NaN                NaN            NaN              NaN   
3                    NaN                NaN            NaN              NaN   
4                    NaN                NaN            NaN              NaN   
...                  ...                ...            ...              ...   
42533                NaN                NaN            NaN              NaN   
42534                NaN                NaN            NaN              NaN   
42535                NaN                NaN            NaN              NaN   
42536                NaN                NaN            NaN              NaN   
42537                NaN                NaN            NaN              NaN   

      pub_rec_bankruptcies tax_liens  tot_hi_cred_lim  total_bal_ex_mort  \
0                      0.0       0.0              NaN                NaN   
1                      0.0       0.0              NaN                NaN   
2                      0.0       0.0              NaN                NaN   
3                      0.0       0.0              NaN                NaN   
4                      0.0       0.0              NaN                NaN   
...                    ...       ...              ...                ...   
42533                  NaN       NaN              NaN                NaN   
42534                  NaN       NaN              NaN                NaN   
42535                  NaN       NaN              NaN                NaN   
42536                  NaN       NaN              NaN                NaN   
42537                  NaN       NaN              NaN                NaN   

      total_bc_limit  total_il_high_credit_limit  
0                NaN                         NaN  
1                NaN                         NaN  
2                NaN                         NaN  
3                NaN                         NaN  
4                NaN                         NaN  
...              ...                         ...  
42533            NaN                         N

Now taking a look at columns with missing values

In [17]:
loans_clean.isna().sum()

loan_amnt                             3
funded_amnt                           3
funded_amnt_inv                       3
term                                  3
int_rate                              3
installment                           3
sub_grade                             3
emp_length                         1115
home_ownership                        3
annual_inc                            7
verification_status                   3
loan_status                           3
pymnt_plan                            3
purpose                               3
title                                16
addr_state                            3
dti                                   3
delinq_2yrs                          32
earliest_cr_line                     32
fico_range_low                        3
fico_range_high                       3
inq_last_6mths                       32
mths_since_last_delinq            26929
mths_since_last_record            38887
open_acc                             32


There are numerous columns full of missing values which should be removed.

In [18]:
loans_clean = loans_clean.dropna(axis = "columns", how = "all")
loans_clean

loan_amnt  funded_amnt  funded_amnt_inv       term int_rate  \
0         5000.0       5000.0           4975.0  36 months   10.65%   
1         2500.0       2500.0           2500.0  60 months   15.27%   
2         2400.0       2400.0           2400.0  36 months   15.96%   
3        10000.0      10000.0          10000.0  36 months   13.49%   
4         3000.0       3000.0           3000.0  60 months   12.69%   
...          ...          ...              ...        ...      ...   
42533     2525.0       2525.0            225.0  36 months    9.33%   
42534     6500.0       6500.0              0.0  36 months    8.38%   
42535     5000.0       5000.0              0.0  36 months    7.75%   
42536        NaN          NaN              NaN        NaN      NaN   
42537        NaN          NaN              NaN        NaN      NaN   

       installment sub_grade emp_length home_ownership  annual_inc  ...  \
0           162.87        B2  10+ years           RENT     24000.0  ...   
1            59.83        C4   < 1 year           RENT     30000.0  ...   
2            84.33        C5  10+ years           RENT     12252.0  ...   
3           339.31        C1  10+ years           RENT     49200.0  ...   
4            67.79        B5     1 year           RENT     80000.0  ...   
...            ...       ...        ...            ...         ...  ...   
42533        80.69        B3   < 1 year           RENT    110000.0  ...   
42534       204.84        A5   < 1 year           NONE         NaN  ...   
42535       156.11        A3  10+ years       MORTGAGE     70000.0  ...   
42536          NaN       NaN        NaN            NaN         NaN  ...   
42537          NaN       NaN        NaN            NaN         NaN  ...   

      last_fico_range_high last_fico_range_low collections_12_mths_ex_med  \
0                    744.0               740.0                        0.0   
1                    499.0                 0.0                        0.0   
2                    719.0               715.0                        0.0   
3                    604.0               600.0                        0.0   
4                    694.0               690.0                        0.0   
...                    ...                 ...                        ...   
42533                714.0               710.0                        NaN   
42534                724.0               720.0                        NaN   
42535                794.0               790.0                        NaN   
42536                  NaN                 NaN                        NaN   
42537                  NaN                 NaN                        NaN   

      policy_code application_type acc_now_delinq  chargeoff_within_12_mths  \
0             1.0       INDIVIDUAL            0.0                       0.0   
1             1.0       INDIVIDUAL            0.0                       0.0   
2             1.0       INDIVIDUAL            0.0                       0.0   
3             1.0       INDIVIDUAL            0.0                       0.0   
4             1.0       INDIVIDUAL            0.0                       0.0   
...           ...              ...            ...                       ...   
42533         1.0       INDIVIDUAL            NaN                       NaN   
42534         1.0       INDIVIDUAL            NaN                       NaN   
42535         1.0       INDIVIDUAL            NaN                       NaN   
42536         NaN              NaN            NaN                       NaN   
42537         NaN              NaN            NaN                       NaN   

       delinq_amnt pub_rec_bankruptcies  tax_liens  
0              0.0                  0.0        0.0  
1              0.0                  0.0        0.0  
2              0.0                  0.0        0.0  
3              0.0                  0.0        0.0  
4              0.0                  0.0        0.0  
...            ...                  ...        ...  
42533          NaN  

In [19]:
loans_clean.iloc[0]

loan_amnt                            5000
funded_amnt                          5000
funded_amnt_inv                      4975
term                            36 months
int_rate                           10.65%
installment                        162.87
sub_grade                              B2
emp_length                      10+ years
home_ownership                       RENT
annual_inc                          24000
verification_status              Verified
loan_status                    Fully Paid
pymnt_plan                              n
purpose                       credit_card
title                            Computer
addr_state                             AZ
dti                                 27.65
delinq_2yrs                             0
earliest_cr_line                 Jan-1985
fico_range_low                        735
fico_range_high                       739
inq_last_6mths                          1
mths_since_last_delinq                NaN
mths_since_last_record            

In [20]:
loans_clean.shape

(42538, 53)

There are still quite a few columns which will be irrelevant in predicting loan defaults so these also need to be removed.  

mths_since_last_record, mths_since_last_delinq and next_pymnt_d also have thousands of missing values and are not relevant to prodicting defaults so will be removed.

The following variables do not give insight into predicting defaults and are also being removed:
funded_amnt, funded_amnt_inv, int_rate, pymnt_plan, last_fico_range_low, last_fico_range_high, collections_12_mths_ex_med, policy_code, acc_now_delinq, delinq_amnt.

In [21]:
loans_clean = loans_clean.drop(["mths_since_last_record", "mths_since_last_delinq", "next_pymnt_d", 
                                "funded_amnt", "funded_amnt_inv", "int_rate", "pymnt_plan", 
                                "mths_since_last_delinq", "collections_12_mths_ex_med", 
                                "policy_code", "acc_now_delinq", "delinq_amnt", "last_fico_range_low", "last_fico_range_high"], 
                               axis = 1)
loans_clean

loan_amnt       term  installment sub_grade emp_length home_ownership  \
0         5000.0  36 months       162.87        B2  10+ years           RENT   
1         2500.0  60 months        59.83        C4   < 1 year           RENT   
2         2400.0  36 months        84.33        C5  10+ years           RENT   
3        10000.0  36 months       339.31        C1  10+ years           RENT   
4         3000.0  60 months        67.79        B5     1 year           RENT   
...          ...        ...          ...       ...        ...            ...   
42533     2525.0  36 months        80.69        B3   < 1 year           RENT   
42534     6500.0  36 months       204.84        A5   < 1 year           NONE   
42535     5000.0  36 months       156.11        A3  10+ years       MORTGAGE   
42536        NaN        NaN          NaN       NaN        NaN            NaN   
42537        NaN        NaN          NaN       NaN        NaN            NaN   

       annual_inc verification_status  \
0         24000.0            Verified   
1         30000.0     Source Verified   
2         12252.0        Not Verified   
3         49200.0     Source Verified   
4         80000.0     Source Verified   
...           ...                 ...   
42533    110000.0        Not Verified   
42534         NaN        Not Verified   
42535     70000.0        Not Verified   
42536         NaN                 NaN   
42537         NaN                 NaN   

                                              loan_status         purpose  \
0                                              Fully Paid     credit_card   
1                                             Charged Off             car   
2                                              Fully Paid  small_business   
3                                              Fully Paid           other   
4                                                 Current           other   
...                                                   ...             ...   
42533  Does not meet the credit policy. Status:Fully Paid           other   
42534  Does not meet the credit policy. Status:Fully Paid           other   
42535  Does not meet the credit policy. Status:Fully Paid           other   
42536                                                 NaN             NaN   
42537                                                 NaN             NaN   

       ... total_rec_late_fee recoveries  collection_recovery_fee  \
0      ...               0.00       0.00                     0.00   
1      ...               0.00     117.08                     1.11   
2      ...               0.00       0.00                     0.00   
3      ...              16.97       0.00                     0.00   
4      ...               0.00       0.00                     0.00   
...    ...                ...        ...                      ...   
42533  ...               0.00       0.00                     0.00   
42534  ...               0.00       0.00                     0.00   
42535  ...               0.00       0.00                     0.00   
42536  ...                NaN        NaN                      NaN   
42537  ...                NaN        NaN                      NaN   

       last_pymnt_d last_pymnt_amnt  last_credit_pull_d  application_type  \
0          Jan-2015          171.62            Sep-2016        INDIVIDUAL   
1          Apr-2013          119.66            Sep-2016        INDIVIDUAL   
2          Jun-2014          649.91            Sep-2016        INDIVIDUAL   
3          Jan-2015          357.48            Apr-2016        INDIVIDUAL   
4          Sep-2016           67.79            Sep-2016        INDIVIDUAL   
...             ...             ...                 ...               ...   
42533      Jun-2010           82.03            May-2007        INDIVIDUAL   
42534      Jun-2010          205.32            Aug-2007        INDIVIDUAL   
42535      Jun-2010          156.39            Feb-2015        INDIVIDUAL   
42536           NaN          

In [22]:
loans_clean.isna().sum()

loan_amnt                      3
term                           3
installment                    3
sub_grade                      3
emp_length                  1115
home_ownership                 3
annual_inc                     7
verification_status            3
loan_status                    3
purpose                        3
title                         16
addr_state                     3
dti                            3
delinq_2yrs                   32
earliest_cr_line              32
fico_range_low                 3
fico_range_high                3
inq_last_6mths                32
open_acc                      32
pub_rec                       32
revol_bal                      3
revol_util                    93
total_acc                     32
initial_list_status            3
out_prncp                      3
out_prncp_inv                  3
total_pymnt                    3
total_pymnt_inv                3
total_rec_prncp                3
total_rec_int                  3
total_rec_

In [23]:
loans_clean.shape

(42538, 40)

In [24]:
#Removing further variables which are irrelevant
loans_clean = loans_clean.drop(["tax_liens", "title", "delinq_2yrs", "inq_last_6mths", "pub_rec", 
                                "pub_rec_bankruptcies", "revol_bal", "revol_util", 
                                "initial_list_status", "out_prncp", "out_prncp_inv", "total_pymnt",
                                "total_pymnt_inv", "total_rec_prncp", "total_rec_int", "total_rec_late_fee",
                                "recoveries", "collection_recovery_fee", "last_pymnt_d", "last_pymnt_amnt",
                                "last_credit_pull_d", "chargeoff_within_12_mths"], axis = 1)
loans_clean

loan_amnt       term  installment sub_grade emp_length home_ownership  \
0         5000.0  36 months       162.87        B2  10+ years           RENT   
1         2500.0  60 months        59.83        C4   < 1 year           RENT   
2         2400.0  36 months        84.33        C5  10+ years           RENT   
3        10000.0  36 months       339.31        C1  10+ years           RENT   
4         3000.0  60 months        67.79        B5     1 year           RENT   
...          ...        ...          ...       ...        ...            ...   
42533     2525.0  36 months        80.69        B3   < 1 year           RENT   
42534     6500.0  36 months       204.84        A5   < 1 year           NONE   
42535     5000.0  36 months       156.11        A3  10+ years       MORTGAGE   
42536        NaN        NaN          NaN       NaN        NaN            NaN   
42537        NaN        NaN          NaN       NaN        NaN            NaN   

       annual_inc verification_status  \
0         24000.0            Verified   
1         30000.0     Source Verified   
2         12252.0        Not Verified   
3         49200.0     Source Verified   
4         80000.0     Source Verified   
...           ...                 ...   
42533    110000.0        Not Verified   
42534         NaN        Not Verified   
42535     70000.0        Not Verified   
42536         NaN                 NaN   
42537         NaN                 NaN   

                                              loan_status         purpose  \
0                                              Fully Paid     credit_card   
1                                             Charged Off             car   
2                                              Fully Paid  small_business   
3                                              Fully Paid           other   
4                                                 Current           other   
...                                                   ...             ...   
42533  Does not meet the credit policy. Status:Fully Paid           other   
42534  Does not meet the credit policy. Status:Fully Paid           other   
42535  Does not meet the credit policy. Status:Fully Paid           other   
42536                                                 NaN             NaN   
42537                                                 NaN             NaN   

      addr_state    dti earliest_cr_line  fico_range_low  fico_range_high  \
0             AZ  27.65         Jan-1985           735.0            739.0   
1             GA   1.00         Apr-1999           740.0            744.0   
2             IL   8.72         Nov-2001           735.0            739.0   
3             CA  20.00         Feb-1996           690.0            694.0   
4             OR  17.94         Jan-1996           695.0            699.0   
...          ...    ...              ...             ...              ...   
42533         NY  10.00              NaN           710.0            714.0   
42534         NY   4.00              NaN           740.0            744.0   
42535         CT   8.81              NaN           770.0            774.0   
42536        NaN    NaN              NaN             NaN              NaN   
42537        NaN    NaN              NaN             NaN              NaN   

       open_acc  total_acc application_type  
0           3.0        9.0       INDIVIDUAL  
1           3.0        4.0       INDIVIDUAL  
2           2.0       10.0       INDIVIDUAL  
3          10.0       37.0       INDIVIDUAL  
4          15.0       38.0       INDIVIDUAL  
...         ...        ...              ...  
42533       NaN        NaN       INDIVIDUAL  
42534       NaN        NaN       INDIVIDUAL  
42535       NaN        NaN       INDIVIDUAL  
42536       NaN        NaN              NaN  
42537       NaN        NaN              NaN  

[42538 rows x 18 columns]